xmtp_mls/src/credential/grant_messaging_access_association.rs

```diff
@@ -0,0 +1,291 @@
+use chrono::DateTime;
+use serde::{Deserialize, Serialize};
+use xmtp_cryptography::signature::{ed25519_public_key_to_address, RecoverableSignature};
+use xmtp_proto::xmtp::mls::message_contents::{
+    GrantMessagingAccessAssociation as GrantMessagingAccessAssociationProto,
+    RecoverableEcdsaSignature as RecoverableEcdsaSignatureProto,
+};
+
+use crate::{
+    types::Address,
+    utils::{address::sanitize_evm_addresses, time::NS_IN_SEC},
+    InboxOwner,
+};
+
+use super::AssociationError;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct UnsignedGrantMessagingAccessData {
+    pub(crate) account_address: Address,
+    pub(crate) installation_public_key: Vec<u8>,
+    pub(crate) created_ns: u64,
+    iso8601_time: String,
```

Comment:
Do we actually need to store the iso8601 time on the struct. Seems like we can just generate it as-needed.

Reply:
The act of generating the iso8601 time can [produce an error](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.from_timestamp), and it is better to validate the incoming data when creating the struct, rather than when it is eventually used to produce `text()`, hence we just generate it up-front